### PR TITLE
Change create daml app to run against several participants

### PR DESCRIPTION
**Pull request by phoebe-da**
I propose to change the `create-daml-app` template so that it can run against a HTTP JSON API port specified in an environment variable, instead of always using port `7575`.The immediate motivation for this is for running `create-daml-app` on Canton: I would like to run several instances of `create-daml-app` against different participants.

Testing:
* Tested manually against `daml-head` on localhost on Chrome for Ubuntu 18, using Sandbox and Canton (following the instructions in the README, using `yarn start` to run in dev mode)
* **Not** tested for DABL, not sure how to do this
* If approved I will test manually on Safari/OSX and (try to) test manually on Edge/Windows

Test recording: https://drive.google.com/file/d/11G9QEVEfEtcHxPwUUuitbw5FDFcOY06x/view?usp=sharing

Showing it running against canton:

image

image

CHANGELOG_BEGIN
Change the `create-daml-app` template so that it can run against a HTTP JSON API port specified in the environment variable `REACT_APP_LEDGER_ID`
CHANGELOG_END

---
### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: templates/create-daml-app/ui/src/config.ts [hunk 1]
```diff
@@ -26,11 +26,3 @@ export const httpBaseUrl =
   deploymentMode === DeploymentMode.PROD_DABL
   ? `https://api.projectdabl.com/data/${ledgerId}/`
   : undefined;
-
-// Unfortunately, the development server of `create-react-app` does not proxy
-// websockets properly. Thus, we need to bypass it and talk to the JSON API
-// directly in development mode.
-export const wsBaseUrl =
```
Comment (phoebe-da) [inline templates/create-daml-app/ui/src/config.ts:33]: I don't think this is needed anymore as the DamlLedger context thing seems to default to where the page is running (normally `localhost::3000`) and the proxy can now be configured to work for websockets

**Comment by phoebe-da** [comment]
lmk if you think the changelog should mention this

**Comment by hurryabit** [comment]
This is truly awesome! I'm very happy to see this ugly hack disappear. Great work. Thank you so much!

I tested if it works on macOS and I am pleased to report it does indeed. And in prod mode as well. Nothing in this PR should affect DABL without also affecting prod mode. Thus, I'm fine with merging this without testing DABL. If the instructions regarding DABL don't work anymore, it's way more likely that DABL has changed since those instructions we're written. I'll take care of checking if the instructions still work myself once this has been merged.

**Comment by phoebe-da** [comment]
works fine on windows 10/edge against daml 1.1.1 (not head since I didn't figure out building the repo from windows)

image

File: templates/create-daml-app/ui/package.json.template [hunk 2]
```diff
@@ -36,13 +36,13 @@
...
-  "proxy": "http://localhost:7575",
   "devDependencies": {
     "@types/jwt-simple": "^0.5.33",
     "@types/react": "^16.9.16",
     "@types/react-dom": "^16.9.4",
     "@types/dotenv": "^8.2.0",
     "react-scripts": "^3.3.0",
-    "typescript": "~3.8.3"
+    "typescript": "~3.8.3",
+    "http-proxy-middleware": "^1.0.4"
```
[the remark this answers is not in the document]
Comment (phoebe-da) [reply templates/create-daml-app/ui/package.json.template]: re-ordered these alphabetically

**Pull request merged**